evalml/pipelines/utils.py

```diff
@@ -188,17 +201,18 @@ def generate_pipeline_code(element):
         pipeline_list += ["{} = '{}'".format(k, v)] if isinstance(v, str) else ["{} = {}".format(k, v)]
 
     pipeline_string = "\t" + "\n\t".join(pipeline_list) + "\n" if len(pipeline_list) else ""
+    _json_dict(element.parameters)
```

Comment:
@bchen1116 could we cut out the middleman by deleting `_json_dict` and doing the following in `generate_pipeline_code`:
```python
try:
    ret = json.dumps(element.parameters, indent='\t')
except TypeError:
    raise TypeError(f"Value {v} of type {type(v)} cannot be JSON-serialized")
```